subsys/bluetooth/host/id.c

```diff
@@ -1969,21 +1971,31 @@ int bt_id_set_adv_own_addr(struct bt_le_ext_adv *adv, uint32_t options,
 			 * problem.
 			 */
 #if defined(CONFIG_BT_OBSERVER)
-			bool scan_enabled = false;
+			bool scan_disabled = false;
```

Comment:
```suggestion
			bool scan_enabled = false;
```

This is the shadow variable to remember that scanning is active. And it needs to be enabled after the address has been updated (after stopping scanning), as Spec does not permit address update if advertising or scanning is active.

Reply:
You are testing `atomic_test_bit(bt_dev.flags, BT_DEV_SCANNING)`  and setting `scan_disabled = true`. How is this readable ?

Reply:
`scan_was_enabled` may be better than `scan_enabled`... 

but again this naming change (commit) is outside scope of this PR, should be a separate PR to make the PR easier to review under its actual scope.

Reply:
> You are testing atomic_test_bit(bt_dev.flags, BT_DEV_SCANNING) and setting scan_disabled = true. How is this readable ?

so if you look on the next line you'll see the scanner is being disabled there. That's all I was trying to express.